rasa/core/training/dsl.py

```diff
@@ -40,13 +44,25 @@ def _parse_item(self, line: Text) -> Optional["Message"]:
             DOCS_BASE_URL
         )
 
-        item_regex = re.compile(r"\s*(.+?):\s*(.*)")
+        # Match three groups:
+        # 1) The correct intent
+        # 2) Optional entities
+        # 3) The message text
+        item_regex = re.compile(r"\s*(.+?)({.*})*:\s*(.*)")
         match = re.match(item_regex, line)
         if match:
```

Comment:
Maybe we could switch this for `if not match:` and raise the exception there, and de-indent the rest of the function.